src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs

```diff
@@ -53,36 +55,51 @@ public sealed class AuthenticationParameters
         /// <summary>
         /// Gets or sets the address of the authority to issue token.
         /// </summary>
-        public string Authority { get; set; }
+        public string Authority { get; }
 
         /// <summary>
         /// Gets or sets the identifier of the target resource that is the recipient of the requested token.
         /// </summary>
-        public string Resource { get; set; }
+        public string Resource { get; }
 
-        private static IHttpManager _httpManager;
+        static AuthenticationParameters()
+        {
+            ModuleInitializer.EnsureModuleInitialized();
+        }
 
-        internal AuthenticationParameters(IHttpManager httpManager)
+        private AuthenticationParameters(string authority, string resource)
         {
-            _httpManager = httpManager;
+            Authority = authority;
+            Resource = resource;
         }
 
         /// <summary>
-        /// Creates authentication parameters from address of the resource. This method expects the resource server to return unauthorized response
-        /// with WWW-Authenticate header containing authentication parameters.
+        /// Sends a GET request to the url provided with no Authenticate header. If a 401 Unauthorized is received, this helper will parse the WWW-Authenticate header to 
+        /// retrieve the authority and resource.
         /// </summary>
         /// <param name="resourceUrl">Address of the resource</param>
         /// <returns>AuthenticationParameters object containing authentication parameters</returns>
-        public async Task<AuthenticationParameters> CreateFromResourceUrlAsync(Uri resourceUrl)
+        /// <remarks>Most protected APIs, including those owned by Microsoft, no longer advertise a resource. Authentication should be done using MSAL, which uses scopes. See https://aka.ms/msal-net-migration-adal-msal </remarks>
```

Comment:
@henrik-me @jmprieur - how about I add these "remarks" to a nice Obsolete message instead? :D

Reply:
Makes sense to me...

---
In reply to: [284784777](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1602#discussion_r284784777) [](ancestors = 284784777)